R-package/tests/testthat/test_helpers.R

```diff
@@ -288,6 +288,72 @@ test_that("xgb.model.dt.tree throws error for gblinear", {
   expect_error(xgb.model.dt.tree(model = bst.GLM))
 })
 
+test_that("xgb.importance works with and without feature names", {
+  importance.Tree <- xgb.importance(feature_names = feature.names, model = bst.Tree)
+  if (!flag_32bit)
+    expect_equal(dim(importance.Tree), c(7, 4))
+  expect_equal(colnames(importance.Tree), c("Feature", "Gain", "Cover", "Frequency"))
+  expect_output(str(importance.Tree), 'Feature.*\\"Age\\"')
+
+  importance.Tree.0 <- xgb.importance(model = bst.Tree)
+  expect_equal(importance.Tree, importance.Tree.0, tolerance = float_tolerance)
+
+  # when model contains no feature names:
+  bst.Tree.x <- bst.Tree
+  bst.Tree.x$feature_names <- NULL
+  importance.Tree.x <- xgb.importance(model = bst.Tree)
+  expect_equal(importance.Tree[, -1, with = FALSE], importance.Tree.x[, -1, with = FALSE],
+               tolerance = float_tolerance)
+
+  imp2plot <- xgb.plot.importance(importance_matrix = importance.Tree)
+  expect_equal(colnames(imp2plot), c("Feature", "Gain", "Cover", "Frequency", "Importance"))
+  xgb.ggplot.importance(importance_matrix = importance.Tree)
+
+  # for multiclass
+  imp.Tree <- xgb.importance(model = mbst.Tree)
+  expect_equal(dim(imp.Tree), c(4, 4))
+
+  trees <- seq(from = 0, by = 2, length.out = 2)
+  importance <- xgb.importance(feature_names = feature.names, model = bst.Tree, trees = trees)
+
+  importance_from_dump <- function() {
+    model_text_dump <- xgb.dump(model = bst.Tree, with_stats = TRUE, trees = trees)
+    imp <- xgb.model.dt.tree(
+      feature_names = feature.names,
+      text = model_text_dump,
+      trees = trees
+    )[
+      Feature != "Leaf", .(
+        Gain = sum(Quality),
+        Cover = sum(Cover),
+        Frequency = .N
+      ),
+      by = Feature
+    ][
+      , `:=`(
+        Gain = Gain / sum(Gain),
+        Cover = Cover / sum(Cover),
+        Frequency = Frequency / sum(Frequency)
+      )
+    ][
+      order(Gain, decreasing = TRUE)
+    ]
+    imp
+  }
+  expect_equal(importance_from_dump(), importance, tolerance = 1e-6)
+
+  ## decision stump
+  m <- xgboost::xgboost(
```

Comment:
I'm not sure if it'd be correct to use `xgboost::` in tests. It's certainly not needed as the whole namespace is imported.